contracts/ConsumerHost.sol

```diff
@@ -256,6 +256,54 @@ contract ConsumerHost is Initializable, OwnableUpgradeable, IConsumer, ERC165 {
         emit Claimed(channelId, consumer, msg.sender, amount, info.balance);
     }
 
+    /**
+     * @notice check the signature from signer or valid consumer
+     * @param channelId the finalized channel ID
+     * @param payload the message signed by sender
+     * @param sign the signature
+     * @return Result of check
+     */
+    function checkSign(uint256 channelId, bytes32 payload, bytes memory sign) external view returns (bool) {
+        bytes32 hash = keccak256(abi.encodePacked('\x19Ethereum Signed Message:\n32', payload));
+        address sConsumer = ECDSA.recover(hash, sign);
+        if (sConsumer == signer) {
+            return true;
+        }
+        return channels[channelId] == sConsumer;
```

Comment:
Is that possible both `channels[channelId]` and `sConsumer` will be `0x00`

Reply:
No, `sConsumer` is recover from ECSDA, if has error in recover, it will panic. And when recover signature, publickey(address) will never be 0x00.